Standards/scs-0302-v2-domain-manager-role.md

```diff
@@ -0,0 +1,143 @@
+by the customer. Only identity resources are owned by domains (with projects
+being also identity resources). Other services use projects for resources
+isolation. They do not need to be domain aware.
+
+Using domains allows implementing [domain
+limits](https://docs.openstack.org/keystone/latest/admin/unified-limits.html#domain-limits)
+which allow to set a global resources limit for the customer. Without domains
+specific limits artificial control of the overall customer consumption must be
+implemented.
+
+#### User management
+
+User management (creation, activation, deactivation, deletion) SHOULD be
+possible using OpenStack APIs.
+
+When an external IdP is being used (IdP federation) there is still an
+expectation that local users may be required by customers. As such, creation of
+users (pre-creating federated users or regular local users) within customer
+domain SHOULD be possible. Keystone does not allow certain operations on
+federated users (i.e. password change, MFA, name) as such allowing customers to
+manage users using OpenStack APIs should not conflict with any additional
+requirements.
+
+Security requirements on the customer side or on the CSP side to only allow
+federated users to consume platform services was used as the limitating factor
+forcing degrading of the capability requirement.
+
+#### Group management
+
+Management of groups SHOULD be possible using OpenStack APIs.
+
+In the case of federated Identities the possibility exists that groups on the IdP
+side do not match groups on the cloud provider side. In addition to that there
+might be a need to combine federated and local users. This would only be
+possible when groups are managed by the OpenStack.
+
+It is advised to keep user groups as mapped entities between external systems
+of CSP and Keystone. Upon user login (or using SCIM) user group relation may be
+synchronized between both platforms.
+
+#### Project deletion
+
+As described above project deletion may be implemented differently by CSPs.
+There are few ways of achieving that:
+
+- Forbid project deletion when resources (i.e. VMs) are still provisioned
+inside of those projects. This scenario assumes that the customer is
+responsible for cleaning projects before their deletion.
+
+- Automatically purge all project resources by the CSP when project deletion
+request is received. In this scenario CSP is implementing custom functionality
+to delete all resources before deleting the project.
+
+- Leave orphaned resources. In this scenario project is being deleted by the API
+with custom cleanup procedures being responsible for dropping orphant resources.
+
+Leaving orphaned resources MAY NOT be allowed.
+
+Forbidding project deletion making customer responsible for the cleanup SHOULD
+be preferred since it allows preventing the accidental deletion of the
+resources. Supplementary methods for purging project resources MAY be offered by
+the CSP.
```

Comment:
To avoid a race condition (be it with checking for existing resources to reject project deletion if there are still resources or be it with deleting the resources automatically), we would need a two phase commit:
(a) mark the project as read-only, so no new resources can be created
(b) do the checks/cleanups
(c) finally remove the project